src/main/java/de/slimecloud/slimeball/features/alerts/youtube/Youtube.java

```diff
@@ -50,11 +56,15 @@ private void check() throws IOException {
 		}
 	}
 
+	private int getApiKey() {
+		return currentKey >= keys.length ? currentKey = 0 : currentKey++;
+	}
+
 	//returns null if no videos found on the channel
 	@Nullable
 	public Video getLastVideo() throws IOException {
 		Request request = new Request.Builder()
-				.url(String.format("https://www.googleapis.com/youtube/v3/search?key=%s&channelId=%s&part=snippet,id&order=date&maxResults=1", authentication, config.getYoutubeChannelId()))
+				.url(String.format("https://www.googleapis.com/youtube/v3/search?key=%s&channelId=%s&part=snippet,id&order=date&maxResults=1", keys[getApiKey()], config.getYoutubeChannelId()))
```

Comment:
Method name is confusing, either make getApiKey actually return the key or rename it so that the name indicates that the current index is returned